cache.go

```diff
@@ -318,8 +432,213 @@ func (c *GoBuildCache) ToRunOption(distroKey string, opts ...GoBuildCacheOption)
+	})
+}
+
+const (
+	sccacheCacheDir    = "/tmp/dalec/sccache-cache"
+	sccacheBinary      = "/tmp/dalec/sccache"
+	sccacheCacheDirWin = "C:\\temp\\dalec\\sccache-cache"
+	sccacheBinaryWin   = "C:\\temp\\dalec\\sccache.exe"
+)
+
+func (c *CargoBuildCache) ToRunOption(worker llb.State, distroKey string, opts ...CargoBuildCacheOption) llb.RunOption {
+	return RunOptFunc(func(ei *llb.ExecInfo) {
+		if c.Disabled {
+			return
+		}
+
+		var info CargoBuildCacheInfo
+		for _, opt := range opts {
+			opt.SetCargoBuildCacheOption(&info)
+		}
+
+		// Ensure platform is set
+		var platform *ocispecs.Platform
+		if ei.Platform != nil {
+			platform = ei.Platform
+		} else if info.Platform != nil {
+			platform = info.Platform
+		} else {
+			p := platforms.DefaultSpec()
+			platform = &p
+		}
+
+		key := fmt.Sprintf("%s-%s-dalec-cargobuildcache", distroKey, platforms.Format(*platform))
+		if c.Scope != "" {
+			key = fmt.Sprintf("%s-%s", key, c.Scope)
+		}
+
+		// Extract distro name for needsPrecompiled check
+		distroName := distroKey
+		if idx := strings.Index(distroKey, "/"); idx != -1 {
+			distroName = distroKey[:idx]
+		}
+
+		// Map version IDs to base names for needsPrecompiled check
+		distroNameMap := map[string]string{
+			"ubuntu22.04": "jammy",
+			"ubuntu20.04": "focal",
+			"ubuntu18.04": "bionic",
+			"debian11":    "bullseye",
+		}
+		if baseName, ok := distroNameMap[distroName]; ok {
+			distroName = baseName
+		}
+
+		// Check if this distro needs precompiled sccache binary or can use package manager
+		needsPrecompiled := map[string]bool{
+			"almalinux8":  true,
+			"almalinux9":  true,
+			"rockylinux8": true,
+			"rockylinux9": true,
+			"bullseye":    true,
+			"bionic":      true,
+			"focal":       true,
+			"jammy":       true,
+		}
+
+		// Determine paths based on platform
+		isWindows := platform.OS == "windows"
+
+		var cacheDir string
+		if isWindows {
+			cacheDir = sccacheCacheDirWin
+		} else {
+			cacheDir = sccacheCacheDir
+		}
+
+		// Set up cache mount for sccache compilation cache
+		llb.AddMount(cacheDir, llb.Scratch(), llb.AsPersistentCacheDir(key, llb.CacheMountShared)).SetRunOption(ei)
+
+		// Set up environment variables
+		llb.AddEnv("SCCACHE_DIR", cacheDir).SetRunOption(ei)
+		llb.AddEnv("SCCACHE_CACHE_SIZE", SccacheCacheSize).SetRunOption(ei)
+
+		// Only download and set up precompiled sccache for distros that need it or Windows
+		if needsPrecompiled[distroName] || isWindows {
+			var binaryPath string
+			if isWindows {
+				binaryPath = sccacheBinaryWin
+			} else {
+				binaryPath = sccacheBinary
+			}
+
+			// Download sccache binary using SourceHTTP during build phase
+			sccacheSource := GetSccacheSource(platform)
+			sccacheState, err := sccacheSource.HTTP.AsState("sccache", llb.Platform(*platform))
+			if err != nil {
+				// If we can't get sccache, continue without it
+				return
+			}
+
+			// Create a setup command to extract and configure sccache
+			var setupCmd string
+			if isWindows {
```

Comment:
We don't build on windows so I think all this can be removed.
Our windows builds are cross compiled from linux.